src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -47,6 +47,25 @@ CORECLR_HOSTING_API(coreclr_initialize,
             void** hostHandle,
             unsigned int* domainId);
 
+//
+// Type of the callback function called by the coreclr_get_error_info
+//
+typedef void (*error_info_callback)(const char* line, void* arg);
+
+//
+// Initialize the CoreCLR. Creates and starts CoreCLR host and creates an app domain
+//
+// Parameters:
+//  callBack                - callback that will be called for each line of the error info
+//  arg                     - argument to pass to the callback
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the error info was successfully retrieved
+//
+CORECLR_HOSTING_API(coreclr_get_error_info,
```

Comment:
This is a big decision. We should include our hosting owners.

/cc @vitek-karas @elinor-fung @agocke